Makefile

```diff
@@ -0,0 +1,6 @@
+COMPOSE := docker-compose
+GO := go
+GOTEST := $(GO) test -race -timeout 10s
+
+test:
+	@$(COMPOSE) up -d elasticsearch localstack && $(COMPOSE) run wait && $(GOTEST) ./...; ret=$$?; $(COMPOSE) down -v && exit $$ret
```

Comment:
What's the story with the `$$?` ?
$$ refers to the current process id
$? refers to the last executed command id (and I think this is the default arg for `exit`)
But what's `$$?` ?

Reply:
Make also use `$` for variables in Makefiles. `$$` is a literal `$`, so the bash subprocess evaluates `ret=$?`.